pkg/internal/validation/dns.go

```diff
@@ -26,9 +26,9 @@ import (
 // "k8s.io/apimachinery" is needed, re-consider whether to add the dependency.
 
 const (
-	dns1123LabelFmt     string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+	dns1123LabelFmt     string = "[a-z0-9](?:[-a-z0-9]*[a-z0-9])?"
 	dns1123SubdomainFmt string = dns1123LabelFmt + "(\\." + dns1123LabelFmt + ")*"
-	dns1035LabelFmt     string = "[a-z]([-a-z0-9]*[a-z0-9])?"
+	dns1035LabelFmt     string = "[a-z](?:[-a-z0-9]*[a-z0-9])?"
```

Comment:
@Adirio, 

For we not add ore one dependency to the project we decided to copy and paste the implementation to validate. 
We should not change this file unless it was changed in the `k8s.io/apimachinery` 

See: https://github.com/kubernetes-sigs/kubebuilder/blob/175952919092cd7304e5e8aeddb9cce37b6a614b/pkg/internal/validation/dns.go#L24-L26

Reply:
Could you please revert the changes here?

Reply:
I would agree with you if the code was copy-pasted, but it has been already modified. You can [check it](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go). So Convering that capture group into a non-capture group tells the future us reading the code that we are not using that capture group to take a substring out of the regex.

Reply:
Could we keep the sample implementation of https://github.com/kubernetes/apimachinery/blob/v0.18.6/pkg/util/validation/validation.go ? (copy and paste)

Reply:
Catcher. I would prefer if possible we keep a copy and paste of https://github.com/kubernetes/apimachinery/blob/v0.18.6/pkg/util/validation/validation.go whcih would allow us to move to use the dependency if we wish without breaking changes. However, I will let it up to you and @ezric